### PR TITLE
[PWX-35366][Oracle] Fixed DevicePath() API for Oracle cloud.

### PR DESCRIPTION
**Pull request by vinayakshnd**
**What this PR does / why we need it**:
Currently, cloudops only looks at the first volumeattachment from list of volumeattachements for a given volume in order to get its device path. But in case of failovers, (ie. existing storage node is deleted), same volume gets attached to new node creating a new volumeattachment which exists along with old volumeattachment for old node (in DETACHED state).

With this new PR, cloudops will go through list of all volumeattachements for a given volume and will use volumeattachement  obj that is in ATTACHED state to find out device path.

**Which issue(s) this PR fixes** (optional)
Closes #
https://portworx.atlassian.net/browse/PWX-35366

**Special notes for your reviewer**:

**Scenario 1: When device is detached.**

Screenshot 2023-12-13 at 7 13 59 PM

```
[opc@oke-cii3qdqu64q-nxlffs5mlra-sraq6gvhj3a-0 oracle]$ go test -v .
=== RUN   TestAll
    cloudops.go:336: 
                Error Trace:    /home/opc/go/src/github.com/libopenstorage/cloudops/oracle/cloudops.go:336
                                                        /home/opc/go/src/github.com/libopenstorage/cloudops/oracle/cloudops.go:59
                                                        /home/opc/go/src/github.com/libopenstorage/cloudops/oracle/oracle_test.go:49
                Error:          Received unexpected error:
                                Volume is detached
                Test:           TestAll
                Messages:       get device path returned error
--- FAIL: TestAll (0.11s)
FAIL
FAIL    github.com/libopenstorage/cloudops/oracle       0.118s
FAIL
```

**Scenario 2: Device attached to an different instance (remote/ instance other than where code is running)**

Screenshot 2023-12-13 at 7 17 19 PM

```
[opc@oke-cii3qdqu64q-nxlffs5mlra-sraq6gvhj3a-0 oracle]$ go test -v .
=== RUN   TestAll
    cloudops.go:336: 
                Error Trace:    /home/opc/go/src/github.com/libopenstorage/cloudops/oracle/cloudops.go:336
                                                        /home/opc/go/src/github.com/libopenstorage/cloudops/oracle/cloudops.go:59
                                                        /home/opc/go/src/github.com/libopenstorage/cloudops/oracle/oracle_test.go:49
                Error:          Received unexpected error:
                                Volume attached on "ocid1.instance.oc1.iad.anuwcljrxypu2kicvjypaug4uimrnqlr4jis7nwg2g6uj4ad2ua3vxp3fc7a" current instance "ocid1.instance.oc1.iad.anuwcljsxypu2kicdj2jz4qvep7sfarg5f5senbzqjlhca7gy7szol4sm3va"
                Test:           TestAll
                Messages:       get device path returned error
--- FAIL: TestAll (0.32s)
FAIL
FAIL    github.com/libopenstorage/cloudops/oracle       0.332s
FAIL
```

**Scenario 3: Device attached on the same/local node where code is running.**

Screenshot 2023-12-13 at 7 26 09 PM

```
[opc@oke-cii3qdqu64q-nxlffs5mlra-sraq6gvhj3a-0 oracle]$ go test -v .
=== RUN   TestAll
time="2023-12-13T13:54:15Z" level=info msg="disk: ocid1.volume.oc1.iad.abuwcljs3m3r5tudbxk5rdkeddni5bk47jrk66fn4jzdbin4lderucxfu4vq device path is: /dev/oracleoci/oraclevdb"
--- PASS: TestAll (0.10s)
PASS
ok      github.com/libopenstorage/cloudops/oracle       0.112s
[opc@oke-cii3qdqu64q-nxlffs5mlra-sraq6gvhj3a-0 oracle]$ 
```

**Scenario 4: Device detached and re-attached with different path. Only the latest volumeattachment with updated devicepath reported.**

Screenshot 2023-12-13 at 7 28 13 PM

```
[opc@oke-cii3qdqu64q-nxlffs5mlra-sraq6gvhj3a-0 oracle]$ go test -v .
=== RUN   TestAll
Running all oracle teststime="2023-12-13T13:59:14Z" level=info msg="Trying to get device path for ocid1.volume.oc1.iad.abuwcljs3m3r5tudbxk5rdkeddni5bk47jrk66fn4jzdbin4lderucxfu4vq\n"
time="2023-12-13T13:59:14Z" level=info msg="disk: ocid1.volume.oc1.iad.abuwcljs3m3r5tudbxk5rdkeddni5bk47jrk66fn4jzdbin4lderucxfu4vq device path is: /dev/oracleoci/oraclevdc"
--- PASS: TestAll (0.13s)
PASS
ok      github.com/libopenstorage/cloudops/oracle       0.146s
```

### CODE REVIEW

File: oracle/oracle.go [hunk 1]
```diff
@@ -376,36 +376,83 @@ func (o *oracleOps) DevicePath(volumeID string) (string, error) {
...
 	}
 
-	volumeAttachment := volumeAttachmentResp.Items[0]
+	var latestVolumeAttachment core.VolumeAttachment
+	for _, va := range volumeAttachmentResp.Items {
+		if va.GetLifecycleState() == core.VolumeAttachmentLifecycleStateAttached {
+			// OCI Block Volumes (cloud-drives) created by PX are non-sharable,
+			// So, at any point in time, only one volumeAttachment will be in ATTACHED state
+			latestVolumeAttachment = va
+			break
```
Comment (nrevanna) [inline oracle/oracle.go:428]: Instead of breaking, we should get all that says "Attached" and if it's attached on more than 1 instance, error out.

Comment (vinayakshnd) [reply oracle/oracle.go:438]: Done!

File: .travis.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ sudo: required
 dist: xenial
 language: go
 go:
-  - 1.17.x
+  - 1.18.x
```
Comment (nrevanna) [inline .travis.yml:5]: Why this bump?

File: oracle/oracle.go [hunk 3]
```diff
@@ -376,36 +376,93 @@ func (o *oracleOps) DevicePath(volumeID string) (string, error) {
...
+				attachedInstanceID = *va.GetInstanceId()
+				latestVolumeAttachment = va
+			} else if attachedInstanceID == *va.GetInstanceId() {
+				latestVolumeAttachment = va
+			} else {
+				return "", cloudops.NewStorageError(cloudops.ErrVolumeAttachedOnMultipleNodes,
+					fmt.Sprintf("Volume %s is attached to multiple instances: [%s] and [%s]",
+						volumeID, attachedInstanceID, *va.GetInstanceId()), "")
+			}
+			break
```
Comment (nrevanna) [inline oracle/oracle.go:438]: This break will exit the loop on finding the first attach. This should not be here

File: .travis.yml (hunk 2 above)
Comment (vinayakshnd) [reply .travis.yml:5]: Our travis CI jobs were failing with below error:
```
make && make test
go build -tags ""  ./...
go vet ./...
GO111MODULE=off go get -u golang.org/x/lint/golint
# golang.org/x/tools/go/ast/astutil
../../../golang.org/x/tools/go/ast/astutil/enclosing.go:323:23: n.Type.TypeParams undefined (type *ast.FuncType has no field or method TypeParams)
```

This `golang.org/x/tools/internal/typeparams` package introduced in go 1.18.
https://pkg.go.dev/golang.org/x/tools/internal/typeparams

After updating go version for travis build, build passed.
https://app.travis-ci.com/github/libopenstorage/cloudops/builds/267823009

File: oracle/oracle.go (hunk 3 above)
Comment (vinayakshnd) [reply oracle/oracle.go]: Done

**Review by nrevanna — APPROVED** [review]
lgtm

**Pull request merged**